r2dbc-spec/src/main/asciidoc/out-parameter-metadata.adoc

```diff
@@ -0,0 +1,81 @@
+[[out-parameter-metadata]]
+= OUT Parameter Metadata
+
+The `OutParametersMetadata` interface is implemented by R2DBC drivers to provide information about OUT parameters.
+It is used primarily by libraries and applications to determine the properties of result parameters.
+
+In cases where the result properties of an SQL statement are unknown until it is run, the `OutParametersMetadata` can be used to determine the actual properties.
+
+`OutParametersMetadata` exposes `OutParameterMetadata` for each returned out parameter.
+Drivers should provide `OutParameterMetadata` on a best-effort basis.
+Out parameter metadata is typically a by-product of stored procedure execution.
+The amount of available information is vendor-dependent.
+Metadata retrieval can require additional lookups (internal queries) to provide a complete metadata descriptor.
+Issuing queries during result processing conflicts with the streaming nature of R2DBC.
+Consequently, `OutParameterMetadata` declares two sets of methods: methods that must be implemented and methods that can optionally be implemented by drivers.
+
+== Obtaining a `OutParametersMetadata` Object
+
+A `OutParametersMetadata` object is created during results consumption through `Result.map(…)`.
+It is created once for all out parameters.
+The following example illustrates retrieval and usage by using an anonymous inner class:
+
+.Using `OutParametersMetadata` and retrieving `OutParameterMetadata`
+====
+[source,java]
+----
+// result is a Result object
+result.map(new Function<Gettable, Object>() {
+
+    @Override
+    public Object apply(Gettable gettable) {
+
+        if (gettable instanceof OutParameters) {
+
+            OutParameters out = (OutParameters) gettable;
+            OutParametersMetadata md = out.getMetadata();
+
+            OutParameterMetadata my_parameter = md.getParameterMetadata("my_parameter");
+            Nullability nullability = my_parameter.getNullability();
+            // …
+        }
+    }
+});
+----
+====
+
+[[outparametermetadata]]
+== Retrieving `OutParameterMetadata`
+
+`OutParameterMetadata` methods are used to retrieve metadata for a single parameter or all parameters.
+
+* `getParameterMetadata(int)` returns the `OutParameterMetadata` by using a zero-based index.
+See <<compliance.guidelines>>.
```

Comment:
I think it's worth documenting here whether or not non-out parameters occupy such indexes or not, i.e. whether indexes of out parameters are consecutive or whether they have gaps, depending on what you've decided.